modules/mbedtls/Kconfig.tls-generic

```diff
@@ -559,6 +559,16 @@ config MBEDTLS_PSA_P256M_DRIVER_ENABLED
 	  Enable support for the optimized sofware implementation of the secp256r1
 	  curve through the standard PSA API.
 
+config MBEDTLS_PSA_KEY_SLOT_COUNT
+	int "Number of key slots in PSA core"
+	default 32
```

Comment:
With this default static key slot count we end up with a relatively fat RAM increase (+2KiB). I get that it's the default in Mbed TLS, but the defaults are not necessarily optimized for embedded devices. Maybe we could make the default such that it's at least not bigger in RAM than with the dynamic key slots? (maybe ~12)

Or the best could be to find the sweet spot to have this be as low as possible to help with the footprint.

Also, will this (key management code) automatically be disabled if no crypto feature requiring keys is enabled?

Reply:
> Also, will this (key management code) automatically be disabled if no crypto feature requiring keys is enabled?

As far as I know the key management code is part of the build even if, for example, you are asking only for some hash support and nothing else. What (I think) it happens is that this block of code gets removed by the linker in that case because no one is making use of it. But this is just my hypothesis, I didn't check in the detail with a real build.

Reply:
Yeah, the linker does remove dead code. My concern was that some compiled-in Mbed TLS code would call this key management code even if no crypto feature requiring keys is enabled, in which case it would prevent the linker from removing that code because it wouldn't be considered dead.

Reply:
Wait, but if "some compiled-in Mbed TLS code" calls some function for key management then I'm tempted to say that it's going to use it for some reason. So it makes sense to keep the functionality. Sure I missed the point here

Reply:
I just remember from when I was looking into reducing the overhead brought by PSA Crypto that some code related to error codes was compiled in (and not removed later) because configuration options were enabled, e.g. ciphers, key types... Even though no operation was made with a particular layer/whatever, its error codes were still used and so this embedded extra code. So I thought of something like that here. But of course I don't know and hopefully it's not the case.